assets/javascripts/modules/ajaxCallbacks.js

```diff
@@ -107,6 +107,113 @@ var ajaxCallbacks = {
+    callbacks: {
+      success: function(response, $element, data, helpers, targets, container, type) {
+        var email = $element.find('[name=email]').val();
+        var $list = $('[data-collaborator-list]');
+        var $tr = $('<tr></tr>').attr('data-collaborator-row', email);
+        var $name = $('<td></td>').text(email).addClass('table--large');
+        var $permission = $('<td></td>').addClass('table--large text--right hard--right');
+        var $action = $('<td></td>').addClass('text--right hard--right');
+        var $form = $('<form></form>')
+          .attr('action', $list.data('collaborator-remove-url') + email)
+          .attr('data-ajax-submit', true)
+          .attr('data-callback-args', '')
+          .attr('data-callback-name', 'apiCollaboratorRemoveResponse.callbacks')
+          .addClass('form');
+        var $span = $('<span></span>')
+            .text('Server error, please try again')
+            .addClass('error-notification js-remove-error');
+        var $button = $('<button type="submit"></button>')
+            .text('Remove')
+            .attr('data-remove-collaborator-link', email)
+            .addClass('button button--link button--small flush hard--right');
+        var $info = $element.find('.js-info');
+        var $message = $info.find('.alert__message');
+        var emailMessage = 'We have sent an email to <strong>'+ email +'</strong>';
+        var message =  emailMessage + (response.registeredUser ?
+                    ' confirming they have been added to this application.' :
+                    ' inviting them to register with the API Developer Hub. They cannot access the application until they register.');
+
+        if ($element.find('[name=role]:checked').val() === 'ADMINISTRATOR') {
+          $permission.append($('<span class="faded-text">Admin</span>'));
+        }
+
+        // add to the list
+        $form.append($span, $button);
+        $action.append($form);
+        $tr.append($name, $permission, $action);
+        $list.append($tr);
+
+        // register new form
+        $(document.body).trigger('registerNewForm', $form);
+
+        // remove any error validation
+        $element.find('.form-field--error').removeClass('form-field--error');
+
+        // display alert info
+        $message.html(message);
+        $info.removeClass('hidden');
+
+        // reset form state & clear value
+        helpers.resetForms(helpers, type, data, container);
+        $element.find('[name=email]').val('');
+
+        helpers.base.success.apply(null, arguments);
+      },
+
+      error: function(response, $element, data, helpers, targets, container, type) {
+        var $email = $element.find('[name=email]');
+        var $error = $email.siblings('.error-notification');
+        var $info = $element.find('.js-info');
+        var $message = $info.find('.alert__message');
+
+        // add error state class
+        $email.parent().addClass('form-field--error');
```

Comment:
current HTML template wraps all input elements inside 'form-field', so it should be fine but Ill add `js-email-field` on the wrapping element